api/lib/mno_enterprise/concerns/controllers/pages_controller.rb

```diff
@@ -59,9 +59,16 @@ def app_logout
   def terms
     @meta[:title] = 'Terms of Use'
     @meta[:description] = 'Terms of Use'
-    ts = MnoEnterprise::App.order_by("updated_at.desc").first.updated_at
-    @apps = Rails.cache.fetch(['pages/terms/app-list', ts]) do
-      MnoEnterprise::App.order_by("name.ac").reject{|i| i.terms_url.blank?}
+
+    ts = MnoEnterprise::App.order_by("updated_at.desc").first
```

Comment:
@ouranos The issue here is when there are no apps. In that case an empty array is returned. That would result in try being called on nil which will throw an error. What if we did it this way? 

`ts = MnoEnterprise::App.order_by("updated_at.desc").first&.updated_at
@apps = if ts
  Rails.cache.fetch(['pages/terms/app-list', ts]) do
    MnoEnterprise::App.order_by("name.ac").reject{|i| i.terms_url.blank?}
  end
else
  []
end`